pykickstart/commands/rhsm.py

```diff
@@ -0,0 +1,118 @@
+        if self.connect_to_insights:
+            retval+=' --connect-to-insights'
+        if self.proxy:
+            retval += ' --proxy="%s"' % self.proxy
+        if self.server_hostname:
+            retval+=' --server-hostname="%s"' % self.server_hostname
+        if self.rhsm_baseurl:
+            retval+=' --rhsm-baseurl="%s"' % self.rhsm_baseurl
+        retval+='\n'
+        return retval
+
+    def _getParser(self):
+        op = KSOptionParser(prog="rhsm", description="""
+                            The rhsm command is used to configure a Red Hat subscription.
+                            A Red Hat subscription is required for installing from
+                            the Red Hat CDN installation source as well connecting the target
+                            system to Red Hat Insights.
+
+                            Subscription attached during the installation will also be available
+                            on the target system after the installation.
+
+                            System Purpose data specified via the syspurpose command may influence
+                            what subscription will be attached to the system. System Purpose data
+                            set via the syspurpose command will override system purpose data attached
+                            to activation keys.
+
+                            An organization id needs to be specified, as well as at least one activation key.
+                            """, version=RHEL8)
+        op.add_argument("--organization", metavar="<organization_name>", version=RHEL8, required=True,
+                        help="Organization id.")
+        op.add_argument("--activation-key", metavar="<activation_key>", action="append", dest="activation_keys",
+                        version=RHEL8, required=True,
+                        help="Activation key. Option can be used multupple time, once per activation key.")
+        op.add_argument("--connect-to-insights", version=RHEL8, action="store_true", default=False,
+                        help="Connect to Red Hat Insights.")
+        op.add_argument("--proxy", version=RHEL8, help="""
+                        Specify an HTTP proxy to use for subscription purposes.
+                        The syntax is::
+
+                        ``--proxy=[protocol://][username[:password]@]host[:port]``
+                        """)
+        op.add_argument("--server-hostname", metavar="<subscription_service_server_url>", version=RHEL8, required=False,
+                        help="Red Hat subscription service server URL.")
+        op.add_argument("--rhsm-baseurl", metavar="<content_base_url>", version=RHEL8, required=False,
+                        help="Content base URL.")
+        return op
+
+    def parse(self, args):
+        (ns, extra) = self.op.parse_known_args(args=args, lineno=self.lineno)
+
+        if len(extra) > 0:
+            msg = _("Unsupported arguments specified for the rhsm command: %s" % extra)
```

Comment:
Move the % extra outside the translation _() brackets.
pylint generates a warning about this, which isn't obvious unless you look at the test run logs.

Reply:
Good catch! :) Fixed.